programs/compressed-token/src/process_compress_spl_token_account.rs

```diff
@@ -0,0 +1,107 @@
+use super::TransferInstruction;
+use crate::{
+    process_transfer::{
+        process_transfer, CompressedTokenInstructionDataTransfer, PackedTokenTransferOutputData,
+    },
+    ErrorCode,
+};
+use anchor_lang::prelude::*;
+use light_system_program::sdk::CompressedCpiContext;
+
+pub fn process_compress_spl_token_account<'info>(
+    ctx: Context<'_, '_, '_, 'info, TransferInstruction<'info>>,
+    owner: Pubkey,
+    remaining_amount: Option<u64>,
+    cpi_context: Option<CompressedCpiContext>,
+) -> Result<()> {
+    let compression_token_account =
+        if let Some(token_account) = ctx.accounts.compress_or_decompress_token_account.as_ref() {
+            token_account
+        } else {
+            return err!(ErrorCode::CompressedPdaUndefinedForCompress);
+        };
+    let compress_amount = compression_token_account
+        .amount
+        .checked_sub(remaining_amount.unwrap_or_default())
+        .ok_or(ErrorCode::ArithmeticUnderflow)?;
```

Comment:
The ErrorCode could be more precise,
Users who put in a remaining_amount that's higher than the available amount should maybe get an error that states insufficient funds or something like that